src/Monotonic.cpp

```diff
@@ -456,25 +615,47 @@ class MonotonicVisitor : public IRVisitor {
     }
 
 public:
-    Monotonic result;
+    ConstantInterval result;
 
-    MonotonicVisitor(const std::string &v, const Scope<Monotonic> &parent)
-        : var(v), result(Monotonic::Unknown) {
+    DerivativeBounds(const std::string &v, const Scope<ConstantInterval> &parent)
+        : var(v), result(ConstantInterval::everything()) {
         scope.set_containing_scope(&parent);
     }
 };
 
 }  // namespace
 
-Monotonic is_monotonic(const Expr &e, const std::string &var, const Scope<Monotonic> &scope) {
+ConstantInterval derivative_bounds(const Expr &e, const std::string &var, const Scope<ConstantInterval> &scope) {
     if (!e.defined()) {
-        return Monotonic::Unknown;
+        return ConstantInterval::everything();
     }
-    MonotonicVisitor m(var, scope);
+    DerivativeBounds m(var, scope);
     e.accept(&m);
     return m.result;
 }
 
+Monotonic is_monotonic(const Expr &e, const std::string &var, const Scope<ConstantInterval> &scope) {
+    if (!e.defined()) {
+        return Monotonic::Unknown;
+    }
+    return to_monotonic(derivative_bounds(e, var, scope));
+}
+
+Monotonic is_monotonic(const Expr &e, const std::string &var, const Scope<Monotonic> &scope) {
+    if (!e.defined()) {
+        return Monotonic::Unknown;
+    }
+    Scope<ConstantInterval> intervals_scope;
+    for (Scope<Monotonic>::const_iterator i = scope.cbegin(); i != scope.cend(); ++i) {
+        intervals_scope.push(i.name(), to_interval(i.value()));
+    }
+    return is_monotonic(e, var, intervals_scope);
+}
+
+Monotonic is_monotonic_strong(const Expr &e, const std::string &var) {
```

Comment:
Not sure I see the difference between `is_monotonic_strong` and `is_monotonic` with a default empty scope argument, but `is_monotonic_strong` doesn't seem to be used anywhere, so could probably be removed?